packages/frontend-shared/src/gql-components/error/BaseError.vue

```diff
@@ -23,8 +23,9 @@
             variant="outline"
             data-testid="error-retry-button"
             :prefix-icon="RestartIcon"
-            prefix-icon-class="icon-dark-indigo-500"
-            @click="retry?.(baseError.id)"
+            :prefix-icon-class="`${retrying && 'animate-spin'} icon-dark-indigo-500`"
+            :disabled="retrying"
+            @click="retryFn()"
```

Comment:
Totally a personal preference, but I prefer to name my event handlers with an `on*` convention so they're more easily recognizable/extensible. In this case, something like `onRetry` or `onRetryClick`